core/src/repair_service.rs

```diff
@@ -315,6 +315,11 @@ impl RepairService {
                             // `slot` is dumped in blockstore wanting to be repaired, we orphan it along with
                             // descendants while copying the weighting heurstic so that it can be
                             // repaired with correct ancestor information
+                            //
+                            // Since this came from replay_stage and these banks were already
+                            // cleared from BankForks which is supposed to be in sync with
+                            // repair_weight we have a guarentee that this is greater than
+                            // `repair_weight.root` which is a pre condition for `split_off`.
```

Comment:
I think you're right, depending on where in the main loop root gets updated we could have a race. It should be safe to just drop dump requests for slots < `repair_weight.root` here, as the `set_root` above would have already purged them and there is no chance that slot specifically would need to be repaired as duplicate or not it could not be part of the rooted path at this point.